packages/components/src/services/hds-time.ts

```diff
@@ -0,0 +1,280 @@
+import Service from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { DateTime } from 'luxon';
+import { isTesting } from '@embroider/macros';
+import type { TaskGenerator } from 'ember-concurrency';
+import type {
+  DisplayType,
+  DefaultDisplayType,
+  DisplayFormatType,
+} from './hds-time-types.ts';
+
+export const MILLISECOND_IN_MS = 1;
+export const SECOND_IN_MS = 1000 * MILLISECOND_IN_MS;
+export const MINUTE_IN_MS = 60 * SECOND_IN_MS;
+export const HOUR_IN_MS = 60 * MINUTE_IN_MS;
+export const DAY_IN_MS = 24 * HOUR_IN_MS;
+export const WEEK_IN_MS = 7 * DAY_IN_MS;
+
+export const THRESHOLD_RELATIVE_TIME_IN_MS = WEEK_IN_MS;
+
+export enum HdsTimeRelativeUnitValues {
+  Second = 'second',
+  Hour = 'hour',
+  Minute = 'minute',
+  Day = 'day',
+  Week = 'week',
+}
+
+export const DEFAULT_RELATIVE_THRESHOLDS = {
+  [HdsTimeRelativeUnitValues.Second]: 1 * MINUTE_IN_MS,
+  [HdsTimeRelativeUnitValues.Minute]: 1 * HOUR_IN_MS,
+  [HdsTimeRelativeUnitValues.Hour]: 1 * DAY_IN_MS,
+  [HdsTimeRelativeUnitValues.Day]: 100 * WEEK_IN_MS,
+};
+
+export enum HdsDisplayKeyValues {
+  FriendlyRelative = 'friendly-relative', // returns 'Sep 5, 2018 (30 minutes ago)'
+  FriendlyLocal = 'friendly-local', // returns 'Sep 5, 2018, 4:07:32 pm'
+  FriendlyOnly = 'friendly-only', // returns 'Sep 5, 2018'
+  Relative = 'relative', // returns 'about 2 hours ago'
+  Utc = 'utc', // returns '2018-09-05T23:15:17345Z'
+}
+
+// // returns 'Sep 5, 2018 (30 minutes ago)'
```

Comment:
to do: remove comments from line 45-58

Reply:
Sorry, I resolved this accidentally & prematurely. I'm updating the comments for clarity but leaving is as I think they serve as a helpful clarification.